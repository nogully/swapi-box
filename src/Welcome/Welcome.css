.Welcome {
  /* Force the body to fill the entire screen */
  width: 100%;
  height: 100%;
  /* Hide elements that flow outside the viewable space */
  overflow: hidden;
  /* Black background for the screen */
}

.fade {
  position: relative;
  width: 100%;
  min-height: 300px;
  top: -25px;
  background-image: linear-gradient(0deg, transparent, black 75%);
  z-index: 1;
}

.star-wars {
  display: flex;
  justify-content: center;
  position: relative;
  height: 300px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 400%;
  font-weight: 600;
  letter-spacing: 1px;
  line-height: 150%;
  perspective: 300px;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 9999px;
  transform-origin: 50% 100%;
  animation: crawl 30s linear;
}

.crawl > .title {
  font-size: 90%;
  text-align: center;
}

.crawl > .title h1 {
  margin: 0 0 100px;
  text-transform: uppercase;
}

@keyframes crawl {
  0% {
    top: 0;
    transform: rotateX(25deg)  translateZ(0);
  }
  100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}