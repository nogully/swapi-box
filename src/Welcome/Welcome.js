import React from 'react';
import './Welcome.css'

const Welcome = (props) => {
  return (
    <div className="Welcome">
    <div className="fade"></div>

    <section className="star-wars">

      <div className="crawl">

        <div className="title">
          <p>EPISODE {props.randomFilm.episode_id}</p>
          <h1>{props.randomFilm.title}</h1>
        </div>
        
        <p>{props.randomFilm.opening_crawl}</p>

      </div>

    </section>
    </div>
  )

}

export default Welcome;