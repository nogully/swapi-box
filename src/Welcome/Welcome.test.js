import React from 'react';
import ReactDOM from 'react-dom';
import Welcome from './Welcome';
import { shallow } from 'enzyme';

describe('Welcome', () => {
  let wrapper 
  let mockFilm

  beforeEach(() => {
    mockFilm = {  "title": "A New Hope", 
                  "episode_id": 4,
                  "opening_crawl": "It is a period of civil war." }
    wrapper = shallow(<Welcome randomFilm={ mockFilm }/>)
  })  

  it('should exist and match snapshot', () => {
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  })

})
