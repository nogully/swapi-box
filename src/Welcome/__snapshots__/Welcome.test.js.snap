// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should exist and match snapshot 1`] = `
<div
  className="Welcome"
>
  <div
    className="fade"
  />
  <section
    className="star-wars"
  >
    <div
      className="crawl"
    >
      <div
        className="title"
      >
        <p>
          EPISODE 
          4
        </p>
        <h1>
          A New Hope
        </h1>
      </div>
      <p>
        It is a period of civil war.
      </p>
    </div>
  </section>
</div>
`;
